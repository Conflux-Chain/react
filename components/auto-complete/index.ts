import AutoComplete, { useAutoCompleteHandle } from './auto-complete'
export { useAutoCompleteHandle }
export default AutoComplete
