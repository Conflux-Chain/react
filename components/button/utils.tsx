import React, { ReactNode } from 'react'
import { NormalSizes } from '../utils/prop-types'
import ButtonIcon from './button-icon'
import { ButtonProps } from './button'
import { ButtonGroupConfig } from '../button-group/button-group-context'

export const getButtonChildrenWithIcon = (
  loading: boolean,
  auto: boolean,
  size: NormalSizes,
  children: ReactNode,
  icons: {
    icon?: React.ReactNode
    iconRight?: React.ReactNode
  },
  isTag: boolean,
) => {
  const { icon, iconRight } = icons
  const hasIcon = icon || iconRight
  const isRight = Boolean(iconRight)
  const paddingForAutoMode =
    auto || size === 'mini' || isTag
      ? `calc(var(--zeit-ui-button-height) / 2 + var(--zeit-ui-button-padding) * .5)`
      : 0
  if (!hasIcon) return <div className={`text ${loading ? 'hidden' : ''}`}>{children}</div>
  return (
    <>
      <ButtonIcon isRight={isRight}>{hasIcon}</ButtonIcon>
      <div className={`text ${isRight ? 'right' : 'left'} ${loading ? 'hidden' : ''}`}>
        {children}
        <style jsx>{`
          .left {
            padding-left: ${paddingForAutoMode};
          }
          .right {
            padding-right: ${paddingForAutoMode};
          }
        `}</style>
      </div>
    </>
  )
}

export const filterPropsWithGroup = (
  props: React.PropsWithChildren<ButtonProps>,
  config: ButtonGroupConfig,
): ButtonProps => {
  if (!config.isButtonGroup) return props
  return {
    ...props,
    auto: true,
    shadow: false,
    ghost: config.ghost || props.ghost,
    size: config.size || props.size,
    variant: config.variant || props.variant,
    color: config.color || props.color,
    disabled: config.disabled || props.disabled,
  }
}
