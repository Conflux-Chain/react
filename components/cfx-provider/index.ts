import CfxProvider from './cfx-provider'

export default CfxProvider
