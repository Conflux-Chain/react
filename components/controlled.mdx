If you think the data flow in react carefully.

uncontrolled component:
caller do not know the internal state of the component

controlled component:
the state of called component is fully controlled by `props` passed in 

The above two are identified by react offical team. However, there is also something 
in between, which can not fit in the patterns mentioned above

third patten: the called need to know the callee state sometims but not always
think form as an example, the caller only need to know the state of callee when 'submit'
happen rather than keep track of the component all the time

forth pattern: the caller need to set the callee state forcelly on a one time, not inter
