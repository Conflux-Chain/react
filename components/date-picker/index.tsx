import { Moment } from 'moment'
import momentGenerateConfig from '@jnoodle/rc-picker/lib/generate/moment'
import generatePicker, {
  PickerProps,
  PickerDateProps,
  RangePickerProps as BaseRangePickerProps,
} from './generate-picker'

export type DatePickerProps = PickerProps<Moment>
export type MonthPickerProps = Omit<PickerDateProps<Moment>, 'picker'>
export type WeekPickerProps = Omit<PickerDateProps<Moment>, 'picker'>
export type RangePickerProps = BaseRangePickerProps<Moment>

export const DatePicker = generatePicker<Moment>(momentGenerateConfig)

export default DatePicker
