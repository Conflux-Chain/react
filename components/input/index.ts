import Input, { useInputHandle } from './input'

export { useInputHandle }
export default Input
