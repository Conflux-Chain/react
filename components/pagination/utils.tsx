export const getPageCount = (total: number, pageSize: number) => {
  return Math.floor((total - 1) / pageSize) + 1
}
