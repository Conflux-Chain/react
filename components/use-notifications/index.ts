import useNotifications from './use-notifications'

export default useNotifications
