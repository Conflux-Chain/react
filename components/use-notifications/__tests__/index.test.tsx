import React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import { useNotifications, CfxProvider } from 'components'
import { nativeEvent, updateWrapper } from 'tests/utils'
import Github from '@zeit-ui/react-icons/github'

const MockNotification: React.FC<{}> = () => {
  const [, setNotification] = useNotifications()
  const clickHandler = (e: any = {}) => {
    const keys = ['title', 'content', 'delay', 'closeable', 'icon', 'placement', 'onClose']
    const params = keys.reduce(
      (pre, key) => {
        const value = e.target[key]
        if (value === undefined) return pre
        return { ...pre, [key]: value }
      },
      { title: '' },
    )
    setNotification(params)
  }
  return (
    <div id="btn" onClick={clickHandler}>
      btn
    </div>
  )
}

const triggerNotification = (wrapper: ReactWrapper, params = {}) => {
  wrapper.find('#btn').simulate('click', {
    ...nativeEvent,
    target: params,
  })
}

const expectNotificationIsShow = (wrapper: ReactWrapper) => {
  const notification = wrapper.find('.notification-container').find('.notification')
  expect(notification.length).not.toBe(0)
}

const expectNotificationIsHidden = (wrapper: ReactWrapper) => {
  const notification = wrapper.find('.notification-container').find('.notification')
  expect(notification.length).toBe(0)
}

describe('UseNotification', () => {
  it('should render correctly', async () => {
    const wrapper = mount(
      <CfxProvider>
        <MockNotification />
      </CfxProvider>,
    )

    expectNotificationIsHidden(wrapper)
    triggerNotification(wrapper, { title: 'test-title', content: 'test-content' })
    await updateWrapper(wrapper, 50)
    expectNotificationIsShow(wrapper)
    expect(wrapper.find('.notification-container').html()).toMatchSnapshot()
  })

  it('should work with different placement', async () => {
    const wrapper = mount(
      <CfxProvider>
        <MockNotification />
      </CfxProvider>,
    )

    expectNotificationIsHidden(wrapper)
    triggerNotification(wrapper, {
      placement: 'right-start',
      title: 'test-title',
      content: 'right-start test-content',
    })
    triggerNotification(wrapper, {
      placement: 'left-start',
      title: 'test-title',
      content: 'left-start test-content',
    })
    await updateWrapper(wrapper, 50)
    expectNotificationIsShow(wrapper)
    expect(wrapper.find('.notification-container').html()).toMatchSnapshot()
  })

  it('should render with custom icon correctly', async () => {
    const wrapper = mount(
      <CfxProvider>
        <MockNotification />
      </CfxProvider>,
    )

    expectNotificationIsHidden(wrapper)
    triggerNotification(wrapper, {
      title: 'test-title',
      content: 'test-content',
      icon: <Github color="red" />,
    })
    await updateWrapper(wrapper, 50)
    expectNotificationIsShow(wrapper)
    expect(wrapper.find('.notification-container').html()).toMatchSnapshot()
  })

  it('should close notification after delay config duration', async () => {
    const wrapper = mount(
      <CfxProvider>
        <MockNotification />
      </CfxProvider>,
    )

    expectNotificationIsHidden(wrapper)
    triggerNotification(wrapper, { delay: 100, title: 'test-title', content: 'test-content' })
    await updateWrapper(wrapper, 50)
    expectNotificationIsShow(wrapper)
    // Element already hidden, but Dom was removed after delay
    await updateWrapper(wrapper, 350)
    const notification = wrapper.find('.notification-container').find('.notification')
    expect(notification.length).toBe(0)
  })

  it('should close notification manually', async () => {
    const wrapper = mount(
      <CfxProvider>
        <MockNotification />
      </CfxProvider>,
    )

    expectNotificationIsHidden(wrapper)
    triggerNotification(wrapper, {
      delay: 0,
      title: 'close manually',
      content: 'test-content',
      closeable: true,
    })
    await updateWrapper(wrapper, 50)
    expectNotificationIsShow(wrapper)
    wrapper.find('.notification').find('.close').at(0).simulate('click', nativeEvent)
    // Element already hidden, but Dom was removed after delay
    await updateWrapper(wrapper, 350)
    const notification = wrapper.find('.notification-container').find('.notification')
    expect(notification.length).toBe(0)
  })

  it('keep display when hover on', async () => {
    const wrapper = mount(
      <CfxProvider>
        <MockNotification />
      </CfxProvider>,
    )

    expectNotificationIsHidden(wrapper)
    triggerNotification(wrapper, {
      delay: 100,
      title: 'display when hover is trigger',
      content: 'test-content',
    })
    await updateWrapper(wrapper, 50)
    expectNotificationIsShow(wrapper)

    wrapper
      .find('.notification-container')
      .find('.notification')
      .simulate('mouseEnter', nativeEvent)
    await updateWrapper(wrapper, 200)

    // Hover event will postpone hidden event
    expectNotificationIsShow(wrapper)
    let notification = wrapper.find('.notification-container').find('.notification')
    expect(notification.length).toBe(1)

    // Restart hidden event after mouse leave
    wrapper
      .find('.notification-container')
      .find('.notification')
      .simulate('mouseLeave', nativeEvent)
    await updateWrapper(wrapper, 350)
    notification = wrapper.find('.notification-container').find('.notification')
    expect(notification.length).toBe(0)
  })

  it('should work with multiple notifications', async () => {
    const wrapper = mount(
      <CfxProvider>
        <MockNotification />
      </CfxProvider>,
    )

    expectNotificationIsHidden(wrapper)
    triggerNotification(wrapper, { delay: 100, title: 'test-title', content: 'test-content' })
    triggerNotification(wrapper, { delay: 100, title: 'test-title', content: 'test-content' })
    triggerNotification(wrapper, { delay: 100, title: 'test-title', content: 'test-content' })
    triggerNotification(wrapper, { delay: 100, title: 'test-title', content: 'test-content' })
    triggerNotification(wrapper, { delay: 100, title: 'test-title', content: 'test-content' })
    triggerNotification(wrapper, { delay: 100, title: 'test-title', content: 'test-content' })

    await updateWrapper(wrapper, 200)
    expectNotificationIsShow(wrapper)
    const visibleNotification = wrapper.find('.notification-container').find('.notification')
    expect(visibleNotification.length).toBe(6)

    await updateWrapper(wrapper, 350)
    const notification = wrapper.find('.notification-container').find('.notification')
    expect(notification.length).toBe(0)
  })

  it('should render with custom icon correctly', async () => {
    const wrapper = mount(
      <CfxProvider>
        <MockNotification />
      </CfxProvider>,
    )

    const handler = jest.fn()
    triggerNotification(wrapper, {
      title: 'test-title',
      content: 'test-content',
      onClose: handler,
      delay: 50,
    })
    expect(handler).not.toHaveBeenCalled()
    await updateWrapper(wrapper, 350)
    expect(handler).toHaveBeenCalled()
  })
})
