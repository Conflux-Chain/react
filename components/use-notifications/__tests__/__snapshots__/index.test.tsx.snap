// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UseNotification should render correctly 1`] = `
"<div class=\\"notification-container\\"><div class=\\"notification  visible  right-start\\"><div class=\\"icon\\"><svg stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" shape-rendering=\\"geometricPrecision\\" viewBox=\\"0 0 24 24\\" height=\\"24\\" width=\\"24\\" style=\\"color: currentColor;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\" fill=\\"currentColor\\"></circle><path stroke=\\"var(--zeit-icons-background)\\" d=\\"M15 9l-6 6M9 9l6 6\\"></path></svg></div><div><div class=\\"title\\">test-title</div><div class=\\"content\\">test-content</div></div><div class=\\"close\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" shape-rendering=\\"geometricPrecision\\" viewBox=\\"0 0 24 24\\" height=\\"24\\" width=\\"24\\" style=\\"color: currentColor;\\"><path d=\\"M18 6L6 18M6 6l12 12\\"></path></svg></div><style>
          .notification {
            max-width: 28rem;
            background-color: #fff;
            color: #333;
            border: 0;
            border-radius: 4px;
            padding: 1.1429rem;
            box-shadow: 0px 6px 16px rgba(0, 0, 0, 0.08);
            margin-top: calc(1.1429rem * 1.5);
            transform: translate(100%, 0);
            opacity: 0;
            transition: transform 150ms, opacity 150ms;
            display: flex;
          }

          .notification.right-start {
            align-self: flex-end;
            margin-right: calc(1.1429rem * 1.25);
          }

          .notification.left-start {
            align-self: flex-start;
            margin-left: calc(1.1429rem * 1.25);
          }

          .notification:first-child {
            margin-top: 0;
          }
          .notification.visible {
            transform: translate(0, 0);
            opacity: 1;
            z-index: 2;
          }
          .notification.hide {
            margin-top: -3.5714rem;
            transform: translate(0, -100%);
            opacity: 0;
            z-index: 1;
          }
          .icon {
            height: 1.4286rem;
            width: 1.4286rem;
            margin-right: calc(1.1429rem * 0.75);
            flex-grow: 1;
            flex-shrink: 0;
            color: #e15c56;
          }
          .icon > :global(svg) {
            width: 100%;
            height: 100%;
          }
          .title {
            font-style: normal;
            font-weight: 500;
            font-size: 1rem;
            line-height: 1.2857rem;
            line-height: 1.4286rem;
          }
          .content {
            line-height: 1.2857rem;
            font-size: 1rem;
            color: #999;
            margin-top: calc(1.1429rem * 0.75);
          }
          .close {
            height: 1.2857rem;
            width: 1.2857rem;
            cursor: pointer;
            margin-left: 1.1429rem;
            color: #999;
            flex-grow: 1;
            flex-shrink: 0;
            margin-right: -0.1429rem;
            margin-top: -0.1429rem;
          }
          .close > :global(svg) {
            width: 100%;
            height: 100%;
          }
        </style></div><style>
        .notification-container {
          position: fixed;
          min-height: 0.7143rem;
          top: calc(1.1429rem * 1.25);
          left: 0;
          right: 0;
          z-index: 2000;
          box-sizing: border-box;
          overflow: visible;
          display: flex;
          flex-direction: column;
          align-items: center;
          height: 0;
          visibility: visible;
        }
      </style></div>"
`;

exports[`UseNotification should render with custom icon correctly 1`] = `
"<div class=\\"notification-container\\"><div class=\\"notification  visible  right-start\\"><div class=\\"icon\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" shape-rendering=\\"geometricPrecision\\" viewBox=\\"0 0 24 24\\" height=\\"24\\" width=\\"24\\" style=\\"color: red;\\"><path d=\\"M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22\\"></path></svg></div><div><div class=\\"title\\">test-title</div><div class=\\"content\\">test-content</div></div><div class=\\"close\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" shape-rendering=\\"geometricPrecision\\" viewBox=\\"0 0 24 24\\" height=\\"24\\" width=\\"24\\" style=\\"color: currentColor;\\"><path d=\\"M18 6L6 18M6 6l12 12\\"></path></svg></div><style>
          .notification {
            max-width: 28rem;
            background-color: #fff;
            color: #333;
            border: 0;
            border-radius: 4px;
            padding: 1.1429rem;
            box-shadow: 0px 6px 16px rgba(0, 0, 0, 0.08);
            margin-top: calc(1.1429rem * 1.5);
            transform: translate(100%, 0);
            opacity: 0;
            transition: transform 150ms, opacity 150ms;
            display: flex;
          }

          .notification.right-start {
            align-self: flex-end;
            margin-right: calc(1.1429rem * 1.25);
          }

          .notification.left-start {
            align-self: flex-start;
            margin-left: calc(1.1429rem * 1.25);
          }

          .notification:first-child {
            margin-top: 0;
          }
          .notification.visible {
            transform: translate(0, 0);
            opacity: 1;
            z-index: 2;
          }
          .notification.hide {
            margin-top: -3.5714rem;
            transform: translate(0, -100%);
            opacity: 0;
            z-index: 1;
          }
          .icon {
            height: 1.4286rem;
            width: 1.4286rem;
            margin-right: calc(1.1429rem * 0.75);
            flex-grow: 1;
            flex-shrink: 0;
            color: #e15c56;
          }
          .icon > :global(svg) {
            width: 100%;
            height: 100%;
          }
          .title {
            font-style: normal;
            font-weight: 500;
            font-size: 1rem;
            line-height: 1.2857rem;
            line-height: 1.4286rem;
          }
          .content {
            line-height: 1.2857rem;
            font-size: 1rem;
            color: #999;
            margin-top: calc(1.1429rem * 0.75);
          }
          .close {
            height: 1.2857rem;
            width: 1.2857rem;
            cursor: pointer;
            margin-left: 1.1429rem;
            color: #999;
            flex-grow: 1;
            flex-shrink: 0;
            margin-right: -0.1429rem;
            margin-top: -0.1429rem;
          }
          .close > :global(svg) {
            width: 100%;
            height: 100%;
          }
        </style></div><style>
        .notification-container {
          position: fixed;
          min-height: 0.7143rem;
          top: calc(1.1429rem * 1.25);
          left: 0;
          right: 0;
          z-index: 2000;
          box-sizing: border-box;
          overflow: visible;
          display: flex;
          flex-direction: column;
          align-items: center;
          height: 0;
          visibility: visible;
        }
      </style></div>"
`;

exports[`UseNotification should work with different placement 1`] = `
"<div class=\\"notification-container\\"><div class=\\"notification  visible  right-start\\"><div class=\\"icon\\"><svg stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" shape-rendering=\\"geometricPrecision\\" viewBox=\\"0 0 24 24\\" height=\\"24\\" width=\\"24\\" style=\\"color: currentColor;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\" fill=\\"currentColor\\"></circle><path stroke=\\"var(--zeit-icons-background)\\" d=\\"M15 9l-6 6M9 9l6 6\\"></path></svg></div><div><div class=\\"title\\">test-title</div><div class=\\"content\\">right-start test-content</div></div><div class=\\"close\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" shape-rendering=\\"geometricPrecision\\" viewBox=\\"0 0 24 24\\" height=\\"24\\" width=\\"24\\" style=\\"color: currentColor;\\"><path d=\\"M18 6L6 18M6 6l12 12\\"></path></svg></div><style>
          .notification {
            max-width: 28rem;
            background-color: #fff;
            color: #333;
            border: 0;
            border-radius: 4px;
            padding: 1.1429rem;
            box-shadow: 0px 6px 16px rgba(0, 0, 0, 0.08);
            margin-top: calc(1.1429rem * 1.5);
            transform: translate(100%, 0);
            opacity: 0;
            transition: transform 150ms, opacity 150ms;
            display: flex;
          }

          .notification.right-start {
            align-self: flex-end;
            margin-right: calc(1.1429rem * 1.25);
          }

          .notification.left-start {
            align-self: flex-start;
            margin-left: calc(1.1429rem * 1.25);
          }

          .notification:first-child {
            margin-top: 0;
          }
          .notification.visible {
            transform: translate(0, 0);
            opacity: 1;
            z-index: 2;
          }
          .notification.hide {
            margin-top: -3.5714rem;
            transform: translate(0, -100%);
            opacity: 0;
            z-index: 1;
          }
          .icon {
            height: 1.4286rem;
            width: 1.4286rem;
            margin-right: calc(1.1429rem * 0.75);
            flex-grow: 1;
            flex-shrink: 0;
            color: #e15c56;
          }
          .icon > :global(svg) {
            width: 100%;
            height: 100%;
          }
          .title {
            font-style: normal;
            font-weight: 500;
            font-size: 1rem;
            line-height: 1.2857rem;
            line-height: 1.4286rem;
          }
          .content {
            line-height: 1.2857rem;
            font-size: 1rem;
            color: #999;
            margin-top: calc(1.1429rem * 0.75);
          }
          .close {
            height: 1.2857rem;
            width: 1.2857rem;
            cursor: pointer;
            margin-left: 1.1429rem;
            color: #999;
            flex-grow: 1;
            flex-shrink: 0;
            margin-right: -0.1429rem;
            margin-top: -0.1429rem;
          }
          .close > :global(svg) {
            width: 100%;
            height: 100%;
          }
        </style></div><div class=\\"notification  visible  left-start\\"><div class=\\"icon\\"><svg stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" shape-rendering=\\"geometricPrecision\\" viewBox=\\"0 0 24 24\\" height=\\"24\\" width=\\"24\\" style=\\"color: currentColor;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\" fill=\\"currentColor\\"></circle><path stroke=\\"var(--zeit-icons-background)\\" d=\\"M15 9l-6 6M9 9l6 6\\"></path></svg></div><div><div class=\\"title\\">test-title</div><div class=\\"content\\">left-start test-content</div></div><div class=\\"close\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" shape-rendering=\\"geometricPrecision\\" viewBox=\\"0 0 24 24\\" height=\\"24\\" width=\\"24\\" style=\\"color: currentColor;\\"><path d=\\"M18 6L6 18M6 6l12 12\\"></path></svg></div><style>
          .notification {
            max-width: 28rem;
            background-color: #fff;
            color: #333;
            border: 0;
            border-radius: 4px;
            padding: 1.1429rem;
            box-shadow: 0px 6px 16px rgba(0, 0, 0, 0.08);
            margin-top: calc(1.1429rem * 1.5);
            transform: translate(-100%, 0);
            opacity: 0;
            transition: transform 150ms, opacity 150ms;
            display: flex;
          }

          .notification.right-start {
            align-self: flex-end;
            margin-right: calc(1.1429rem * 1.25);
          }

          .notification.left-start {
            align-self: flex-start;
            margin-left: calc(1.1429rem * 1.25);
          }

          .notification:first-child {
            margin-top: 0;
          }
          .notification.visible {
            transform: translate(0, 0);
            opacity: 1;
            z-index: 2;
          }
          .notification.hide {
            margin-top: -3.5714rem;
            transform: translate(0, -100%);
            opacity: 0;
            z-index: 1;
          }
          .icon {
            height: 1.4286rem;
            width: 1.4286rem;
            margin-right: calc(1.1429rem * 0.75);
            flex-grow: 1;
            flex-shrink: 0;
            color: #e15c56;
          }
          .icon > :global(svg) {
            width: 100%;
            height: 100%;
          }
          .title {
            font-style: normal;
            font-weight: 500;
            font-size: 1rem;
            line-height: 1.2857rem;
            line-height: 1.4286rem;
          }
          .content {
            line-height: 1.2857rem;
            font-size: 1rem;
            color: #999;
            margin-top: calc(1.1429rem * 0.75);
          }
          .close {
            height: 1.2857rem;
            width: 1.2857rem;
            cursor: pointer;
            margin-left: 1.1429rem;
            color: #999;
            flex-grow: 1;
            flex-shrink: 0;
            margin-right: -0.1429rem;
            margin-top: -0.1429rem;
          }
          .close > :global(svg) {
            width: 100%;
            height: 100%;
          }
        </style></div><style>
        .notification-container {
          position: fixed;
          min-height: 0.7143rem;
          top: calc(1.1429rem * 1.25);
          left: 0;
          right: 0;
          z-index: 2000;
          box-sizing: border-box;
          overflow: visible;
          display: flex;
          flex-direction: column;
          align-items: center;
          height: 0;
          visibility: visible;
        }
      </style></div>"
`;
